src/combinatorics/simple_graph/connectivity.lean

```diff
@@ -1513,6 +1513,84 @@ by { rw ← set.nonempty_iff_univ_nonempty, exact hconn u v }
 lemma connected.set_univ_walk_nonempty (hconn : G.connected) (u v : V) :
   (set.univ : set (G.walk u v)).nonempty := hconn.preconnected.set_univ_walk_nonempty u v
 
+/-! ### Walks as subgraphs -/
+
+namespace walk
+variables {G G'} {u v w : V}
+
+/-- The subgraph consisting of the vertices and edges of the walk. -/
+@[simp] protected def to_subgraph : Π {u v : V}, G.walk u v → G.subgraph
+| u _ nil := G.singleton_subgraph u
+| _ _ (cons h p) := G.subgraph_of_adj h ⊔ p.to_subgraph
+
+lemma to_subgraph_cons_nil_eq_subgraph_of_adj (h : G.adj u v) :
+  (cons h nil).to_subgraph = G.subgraph_of_adj h :=
+by simp
+
+@[simp] lemma verts_to_subgraph (p : G.walk u v) : p.to_subgraph.verts = {w | w ∈ p.support} :=
+begin
+  ext w,
+  induction p with _ x y z h p' ih,
+  { simp },
+  { have : w = y ∨ w ∈ p'.support ↔ w ∈ p'.support :=
+      ⟨by rintro (rfl | h); simp [*], by simp { contextual := tt}⟩,
+    simp [ih, or_assoc, this] }
+end
+
+lemma mem_verts_to_subgraph (p : G.walk u v) :
+  w ∈ p.to_subgraph.verts ↔ w ∈ p.support :=
+by { rw [verts_to_subgraph], refl }
+
+lemma nonempty_verts (p : G.walk u v) : p.to_subgraph.verts.nonempty :=
```

Comment:
This has no preconditions and `set.nonempty` is a precondition to quite a few simp lemmas, so I think this makes sense.
```suggestion
@[simp] lemma nonempty_verts (p : G.walk u v) : p.to_subgraph.verts.nonempty :=
```